central/complianceoperator/v2/pruner/pruner.go

```diff
@@ -29,29 +39,59 @@ type Pruner interface {
 }
 
 // New returns on instance of Manager interface that provides functionality to process compliance requests and forward them to Sensor.
-func New(integrationDS compIntegration.DataStore, scanSettingDS compScanSetting.DataStore, scanResultDS scanResult.DataStore) Pruner {
+func New(integrationDS compIntegration.DataStore, scanSettingDS compScanSetting.DataStore, scanResultDS scanResult.DataStore, compRuleDS compRuleDS.DataStore, profileDS profileDS.DataStore, scanSettingsBindingsDS scanSettingBindingDS.DataStore, scanDS scanDS.DataStore, suitesDS suitesDS.DataStore) Pruner {
 	return &pruneImpl{
-		integrationDS: integrationDS,
-		scanSettingDS: scanSettingDS,
-		scanResultDS:  scanResultDS,
+		integrationDS:          integrationDS,
+		scanSettingDS:          scanSettingDS,
+		scanResultDS:           scanResultDS,
+		compRuleDS:             compRuleDS,
+		profileDS:              profileDS,
+		scanSettingsBindingsDS: scanSettingsBindingsDS,
+		scanDS:                 scanDS,
+		suitesDS:               suitesDS,
 	}
 }
 
 // RemoveComplianceResourcesByCluster removes orphaned compliance operator data for the cluster
 func (p *pruneImpl) RemoveComplianceResourcesByCluster(ctx context.Context, clusterID string) {
 	// Remove the compliance integrations
 	if err := p.integrationDS.RemoveComplianceIntegrationByCluster(ctx, clusterID); err != nil {
-		log.Errorf("failed to delete compliance integration for cluster %q: %v", clusterID, err)
+		log.Errorf("failed to delete compliance integrations for cluster %q: %v", clusterID, err)
 	}
 
 	// Remove any scan configurations for the cluster
 	if err := p.scanSettingDS.RemoveClusterFromScanConfig(ctx, clusterID); err != nil {
-		log.Errorf("failed to delete scan config for cluster %s: %v", clusterID, err)
+		log.Errorf("failed to delete scan configs for cluster %s: %v", clusterID, err)
 	}
 
 	// Remove any scan result for the cluster
 	if err := p.scanResultDS.DeleteResultsByCluster(ctx, clusterID); err != nil {
-		log.Errorf("failed to delete scan result for cluster %s: %v", clusterID, err)
+		log.Errorf("failed to delete scan results for cluster %s: %v", clusterID, err)
+	}
+
+	// Remove any rule for the cluster
+	if err := p.compRuleDS.DeleteRulesByCluster(ctx, clusterID); err != nil {
```

Comment:
nit: renaming `DS` to `DataStore`. It was confusing to read because `ds` is also commonly used for `daemonset`.
In general I prefer not using too many abbreviations because when I follow code pathes I am able to understand the code faster.